File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItWarningLinkParcelable.java
package it.trade.android.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import it.trade.model.reponse.WarningLink;

public class TradeItWarningLinkParcelable implements Parcelable {
    protected String label;
    protected String url;

    TradeItWarningLinkParcelable(WarningLink link) {
        this.label = link.label;
        this.url = link.url;
    }

    public String getLabel() {
        return label;
    }

    public String getUrl() {
        return url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItWarningLinkParcelable that = (TradeItWarningLinkParcelable) o;

        if (label != null ? !label.equals(that.label) : that.label != null) return false;
        return url != null ? url.equals(that.url) : that.url == null;
    }

    @Override
    public int hashCode() {
        int result = label != null ? label.hashCode() : 0;
        result = 31 * result + (url != null ? url.hashCode() : 0);
        return result;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.label);
        dest.writeString(this.url);
    }

    protected TradeItWarningLinkParcelable(Parcel in) {
        this.label = in.readString();
        this.url = in.readString();
    }

    public static final Parcelable.Creator<TradeItWarningLinkParcelable> CREATOR = new Parcelable.Creator<TradeItWarningLinkParcelable>() {
        @Override
        public TradeItWarningLinkParcelable createFromParcel(Parcel source) {
            return new TradeItWarningLinkParcelable(source);
        }

        @Override
        public TradeItWarningLinkParcelable[] newArray(int size) {
            return new TradeItWarningLinkParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/TradeItWarningLinkParcelable.kt
package it.trade.android.sdk.model

import android.os.Parcel
import android.os.Parcelable

import it.trade.model.reponse.WarningLink

class TradeItWarningLinkParcelable : Parcelable {
    var label: String
        protected set
    var url: String
        protected set

    internal constructor(link: WarningLink) {
        this.label = link.label
        this.url = link.url
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TradeItWarningLinkParcelable

        if (label != other.label) return false
        if (url != other.url) return false

        return true
    }

    override fun hashCode(): Int {
        var result = label.hashCode()
        result = 31 * result + url.hashCode()
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.label)
        dest.writeString(this.url)
    }

    protected constructor(`in`: Parcel) {
        this.label = `in`.readString()
        this.url = `in`.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<TradeItWarningLinkParcelable> = object : Parcelable.Creator<TradeItWarningLinkParcelable> {
            override fun createFromParcel(source: Parcel): TradeItWarningLinkParcelable {
                return TradeItWarningLinkParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItWarningLinkParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
